htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1306,9 +1311,11 @@ function generate($format='ALL')
 			// SEPA Initialisation
 			$CrLf = "\n";
 			$date_actu = dol_now();
+			$dateTime_ECMA = dol_print_date($date_actu, '%Y-%m-%dT%H:%M:%S');
+
+                        if(!empty($executiondate)) $date_actu=$executiondate; 
```

Comment:
Are you sure the 
 if(!empty($executiondate)) $date_actu=$executiondate; 
is after 
$dateTime_ECMA = dol_print_date($date_actu, '%Y-%m-%dT%H:%M:%S');
and not before ?

Reply:
Yes. The $dateTime_ECMA is used for the create date and not for execution date. Is only the date who stay the original date.